src/Neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -231,11 +231,14 @@ protected internal void RuntimeLoadScript(byte[] script, CallFlags callFlags, Ar
         /// <returns><see langword="true"/> if the account has witnessed the current transaction; otherwise, <see langword="false"/>.</returns>
         protected internal bool CheckWitness(byte[] hashOrPubkey)
         {
+            // If we don't have the ScriptContainer, we consider that there are no script hashes for verifying
+            if (ScriptContainer is null) throw new ArgumentException("Invalid ScriptContainer", nameof(ScriptContainer));
+
             UInt160 hash = hashOrPubkey.Length switch
             {
                 20 => new UInt160(hashOrPubkey),
                 33 => Contract.CreateSignatureRedeemScript(ECPoint.DecodePoint(hashOrPubkey, ECCurve.Secp256r1)).ToScriptHash(),
-                _ => throw new ArgumentException(null, nameof(hashOrPubkey))
+                _ => throw new ArgumentException("Invalid hashOrPubKey", nameof(hashOrPubkey))
```

Comment:
We can merge this change

Reply:
will crate another pr, this one is mainly about the exception.